ibc-core/ics02-client/src/handler/create_client.rs

```diff
@@ -26,11 +24,13 @@ where
     // Construct this client's identifier
     let id_counter = ctx.client_counter()?;
 
-    let client_state = ctx.decode_client_state(client_state)?;
+    let client_val_ctx = ctx.get_client_validation_context();
```

Comment:
Insightful question!

Currently, the approach is modular, which is especially beneficial for hosts like the Sovereign SDK, allowing the organization of each IBC layer as an independent module, which enhances maintainability, encapsulation, and overall control over components, enabling the assembly of a variety of modules. In addition, this does not enforce the necessity to implement client contexts directly on the main context object too, whereas having the `ClientValidationContext` as a super trait does enforce this requirement. These two objects can live as separate objects, which is a better implementation method IMO. Of course, we can still implement all context traits on a single object, as it is the case with the `MockContext` in the `ibc-testkit`.

Reply:
If we want to maintain `ValidationContext: ClientValidationContext,` it makes sense to merge these two interfaces. IMO this is adding confusion in the interface.

For modularity, let's say for cw-contracts we can implement `ClientExecutionContext` for `OurContext`. But if one wants an `ExecutionContext` on top of this, they can extend the existing `OurContext` instead of thinking of it as a separate object.

Reply:
Ahh, I think I've found the source of confusion. 
Firstly, we don't have  `ValidationContext: ClientValidationContext` at the ICS-24 level. A similar bound exists between ICS-02 and ICS-07 contexts, which makes sense there. As these contexts are not expected to be implemented on different `ClientState` objects. However, we should keep these traits separate due to the Tendermint-specific methods residing under the ICS-07 `ValidationContext`.